### PR TITLE
Address input fixes

### LINKED ISSUE #19427: Paste button not disappearing when data exists in "sent to" field
### Steps:
1. Go to wallet 
2. Open "sent to" page
3. Enter or paste any data into the "sent to" field
4. Tap somewhere else to close the keyboard.
5. Focus on the "sent to" field again

### Actual result:
The 'Paste' button does not disappear.

https://github.com/status-im/status-mobile/assets/52490791/37e86d75-5aff-4ae0-9c95-58014dcea5cf

### Expected result:
The 'paste' button should disappear if some data is present in the "sent to" field
https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=689-192881&mode=design&t=iDfSxwGkN1LYSEf1-0

### ENV:
[28 Mar 2024]

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/19427, https://github.com/status-im/status-mobile/issues/20014

### Summary 
This PR fixes 3 issues with the address input:

1. Paste button not disappearing when there is text
2. X button not appearing after focusing on text
3. Correct check mark disappearing after pasting a correct address, then focusing, then bluring

### Demo

https://github.com/status-im/status-mobile/assets/29354102/75af6daa-59fd-4bd5-837a-287bc0a3b896

### CODE REVIEW

File: src/quo/components/inputs/address_input/view.cljs [hunk 1]
```diff
@@ -132,8 +132,7 @@
        :on-focus               on-focus
        :on-blur                on-blur
        :on-change-text         on-change}]
-     (when (or (= status :default)
-               (= status :active))
+     (when (empty? value)
```
Comment (OmarBasem) [inline src/quo/components/inputs/address_input/view.cljs:136]: First fix (paste button)

File: src/quo/components/inputs/address_input/view.cljs [hunk 2]
```diff
@@ -154,7 +153,9 @@
             :inner-style         (style/accessory-button blur? theme)
             :on-press            on-scan}
            :main-icons/scan])])
-     (when (= status :typing)
+     (when (or (= status :typing)
+               (and (= status :active)
+                    (not-empty value)))
```
Comment (OmarBasem) [inline src/quo/components/inputs/address_input/view.cljs:159]: Second fix (X button)

File: src/quo/components/inputs/address_input/view.cljs [hunk 3]
```diff
@@ -167,7 +168,9 @@
         {:style               style/buttons-container
          :accessibility-label :loading-button-container}
         [loading-icon blur? theme]])
-     (when (and (= status :loading) valid-ens-or-address?)
+     (when (and (or (= status :loading)
+                    (= status :default))
+                valid-ens-or-address?)
```
Comment (OmarBasem) [inline src/quo/components/inputs/address_input/view.cljs]: Third fix (checkmark)

**Review by ulisesmac — APPROVED** [review]
Thanks for the PR, looks good

**Review by J-Son89 — APPROVED** [review]

**Review by briansztamfater — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @OmarBasem !
Thanks for your fix, looks good on iOS.

But here are some Android issues👇

**ISSUE 1:** cross icon doesn't delete entered address on Send to screen

**Steps:**

1. Login to app
2. Go to Wallet -> Account -> Send
3. Type any data into address field
4. Tap on cross

**Expected result:** data is deleted successfully, Past and scan icons appeared
**Actual result:** cross icon doesn't work, data is not deleted and after few attempts it somehow deletes and green checkmark appears near scan icon.
Screenshot 2024-05-17 at 16 21 57


https://github.com/status-im/status-mobile/assets/147824399/7673d5c5-587d-4cd7-a69a-1f71fd544827

**Comment by OmarBasem** [comment]
Thanks for your testing @mariia-skrypnyk! Fixed the issue.

**Comment by mariia-skrypnyk** [comment]
Hi @OmarBasem !

Thanks for handling this issue 🙌!
Looks good now.

Failed e2e are not related, please go ahead and do the merge!

**Pull request merged**